src/csharp/Microsoft.Spark/Utils/UdfSerDe.cs

```diff
@@ -264,7 +264,7 @@ private static TypeData SerializeType(Type type)
                 {
                     Type type = AssemblyLoader.LoadAssembly(
                         td.AssemblyName,
-                        td.AssemblyFileName).GetType(td.Name);
+                        td.AssemblyFileName)?.GetType(td.Name);
```

Comment:
Maybe we can also use the `GetType` method that throws on error as well.
```suggestion
                        td.AssemblyFileName)?.GetType(td.Name, true);
```